### ISSUE TITLE
Misleading "firewall" toast notifications when there is no internet

### ISSUE THREAD

**Issue by knguyenBC** [bcgov/bc-wallet-mobile#1907]
### Issue
The "firewall" toast notification appears randomly at app start. People are confused when they see that notification and are concerned that their app is not working. 

### Acceptance Criteria
- Firewall toast notification should only appear is there is a firewall
- A "no internet" notification should appear if there is no internet
- Add a slight delay in the notification to ensure that the notifications don't appear for just a momentary blip. 
- Update the list of ledger genesis transactions

### UI AC
No internet connection
- text: "no internet connection"
- Button: "X" button that dismisses the toast notification
- Gestures: swipe up to dismiss the toast notification
- Component: use info box, type: error

### Notes
 - Add a flag for enabling/disabling "firewall check". Enabled by default on Bifold. Disabled on BC Wallet.

### Related Issues
- #1829
- #1772

**Comment by knguyenBC** [comment]
Estimated at 5

**Comment by jleach** [comment]
Tried a few things. Updating the ledger nodes (genesis block) and manually confirming the results from some BC Wallet debugging. The wallet is getting a "timeout" on most of the ledger nodes. There are one or two that work and this is probably why the test works but its not overly reliable because if that one node can not or does not respond then the firewall message appears. In any case, we need to find a better way to check a node than opening a TCP conneciton to it on the client_ip and client_port.

```console
 LOG  connected: 154.0.164.39 9702
 LOG  timeout: 54.207.36.81 9702
 LOG  timeout: 128.130.204.35 9722
 LOG  timeout: 35.167.133.255 9702
 LOG  timeout: 34.226.105.29 9701
 LOG  timeout: 34.224.255.108 9796
 LOG  timeout: 38.70.17.248 9702
```

Confirmed with `nmap -Pn -sT --reason -p 9702 54.207.36.81`

```console
Scanning 54.207.36.81 on ports 9702
Command: nmap -Pn -sT --reason -p 9702 54.207.36.81
Starting Nmap 7.95 ( https://nmap.org ) at 2024-06-19 11:04 PDT
Nmap scan report for ec2-54-207-36-81.sa-east-1.compute.amazonaws.com (54.207.36.81)
Host is up, received user-set.

PORT     STATE    SERVICE REASON
9702/tcp filtered unknown no-response
```

We'll talk about adding a proxy so we can use HTTP/S and not direct ZMQ ledger comms in the upcoming spring. Moving back to "Needs Refinement".

**Comment by jleach** [comment]
Moving this in to in-progress for better visibility. This may not be an issue once we start using the VDR proxy. 

Ref. #2089

**Comment by jleach** [comment]
These changes are done but don't work as expected. The call to `const poolTxn = await poolService.getAllPoolTransactions()` returns all txn not just the current state of the nodes which means I have a pretty good chance of getting inactive nodes resulting in false positives indicating a firewall may be blocking traffic. We decided to part this feature for now while we ask the IndyVDR team if it is possible to get active nodes and also because BC Wallet is looking at moving to to the IndyVDR Proxy.

**Comment by knguyenBC** [comment]
@jleach Can we match the no internet connection language and UI? i.e. error toast notification and content being "No internet connection" with a close button? The content that's displayed now is too wordy for a toast notification. Plus, we provide a popup with more information on the impacts of no internet when they try to use a feature that requires it (ex. scanning a qr code)  

![Image](https://github.com/user-attachments/assets/99f52f00-4229-48fe-a7ab-729b83c5c16d)

**Comment by bryce-mcmath** [comment]
Majority of the work is done, but looks like some content-y changes have been requested from the UX team, moving back to Ready

**Comment by knguyenBC** [comment]
added more design specs in AC

**Comment by tom11-nguyen** [comment]
Tested and confirmed there is a close (x) icon inside the "No internet connection" toast.

**Comment by jeznorth** [comment]
Verified fixed.
